src/controllers/TalksController.php

```diff
@@ -498,12 +498,88 @@ protected function getTalkDataFromRequest(PDO $db, Request $request, $event_id)
             FILTER_SANITIZE_URL
         );
 
-        $talk['speakers'] = array_map(function ($speaker) {
-            $speaker = filter_var($speaker, FILTER_SANITIZE_STRING);
-            $speaker = trim($speaker);
-            return $speaker;
-        }, (array) $request->getParameter('speakers'));
+        $talk['speakers'] = array_map(
+            function ($speaker) {
+                $speaker = filter_var($speaker, FILTER_SANITIZE_STRING);
+                $speaker = trim($speaker);
+                return $speaker;
+            },
+            (array) $request->getParameter('speakers')
+        );
+
+        return $talk;
+    }
+
+    public function getSpeakersForTalk(Request $request, PDO $db)
+    {
+        $talk_id = $this->getItemId($request);
+        $talk = $this->getTalkById($db, $request, $talk_id);
+        return $talk->speakers;
+    }
+
+    public function setSpeakerForTalk(Request $request, PDO $db)
+    {
+        //We must be logged in
+        if (!isset($request->user_id)) {
+            throw new Exception("You must be logged in to create data", 401);
+        }
+
+        $talk_id = $this->getItemId($request);
+        $talk_mapper = new TalkMapper($db, $request);
+        $talk = $talk_mapper->getTalkById($talk_id);
+        if (! $talk) {
+            throw new Exception("Talk not found", 404);
```

Comment:
I'm in two minds about this - I totally get where you are coming from, but in 99% of cases the API throws `\Exception` - I don't think it's a policy per-sey, more that they are not really 'meant' to be caught - the handle_exception catches all of them and then turns it into an API response.

That said, we have been discussing the need to standardise the HTTP codes we return, so perhaps that could be tied in with creating our own Exception classes, where the HTTP code is pre-defined as part of the exception. 

This would probably be a separate scope to this PR, but I'll look at raising a separate PR to do it.